lib/modules/dosomething/dosomething_api/resources/signup_resource.inc

```diff
@@ -0,0 +1,113 @@
+<?php
+
+function _signup_resource_definition() {
+  $signup_resource = [];
+  $signup_resource['signup'] = [
```

Comment:
make it plural! all endpoints are best to be kept plural as a general rule for REST APIs and standard to follow :)